ray-operator/apis/ray/v1/raycluster_types.go

```diff
@@ -140,6 +140,14 @@ type RayClusterStatus struct {
 	Head HeadInfo `json:"head,omitempty"`
 	// Reason provides more information about current State
 	Reason string `json:"reason,omitempty"`
+
+	// Represents the latest available observations of a deployment's current state.
+	// +patchMergeKey=type
```

Comment:
Why do we need these?

Reply:
This is used by strategic merge patch: https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/#notes-on-the-strategic-merge-patch.
This is also used by server-side apply, when `listType=map` and `listMapKey` are not specified: https://kubernetes.io/docs/reference/using-api/server-side-apply/#merge-strategy.

Reply:
Yes! Thank you @astefanutti for helping me clarify those markers.